sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/models/CloudServiceConfiguration.java

```diff
@@ -94,4 +95,56 @@ public CloudServiceConfiguration setOsVersion(String osVersion) {
         this.osVersion = osVersion;
         return this;
     }
+
+    /**
+     * Creates a representation of this object as JSON.
+     *
+     * @param jsonWriter Where the JSON representation is written to.
+     *
+     * @throws IOException Error while writing to JsonWriter.
+     * @return The JsonWriter with the JSON representation of this object.
+     */
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        jsonWriter.writeStartObject();
+        jsonWriter.writeStringField("osFamily", osFamily);
+        jsonWriter.writeStringField("osVersion", osVersion);
+        jsonWriter.writeEndObject();
+
+        return jsonWriter;
+    }
+
+    /**
+     * Creates a new instance from its JSON representation.
+     *
+     * @param jsonReader Where the JSON read from to deserialize the object.
+     *
+     * @throws IOException Error while reading from JsonReader.
+     * @return Instance read from the JsonReader.
+     */
+    public static CloudServiceConfiguration fromJson(JsonReader jsonReader) throws IOException {
+        return jsonReader.readObject(reader -> {
+            String family = null;
+            String version = null;
+
+            while (reader.nextToken() != JsonToken.END_OBJECT) {
+                String fieldName = reader.getFieldName();
+                reader.nextToken();
+
+                if ("osFamily".equals(fieldName)) {
+                    family = reader.getString();
+                } else if ("osVersion".equals(fieldName)) {
+                    version = reader.getString();
+                } else {
+                    reader.skipChildren();
+                }
+            }
+
+            if (family != null) {
+                return new CloudServiceConfiguration(family).setOsVersion(version);
```

Comment:
Will this be a potential regression? What would have happened previously if a json payload somehow didn't have a family?

Reply:
Urgh. Good find! :(